code/game/objects/items/devices/floor_painter.dm

```diff
@@ -0,0 +1,201 @@
+// Floor painter
+
+/obj/item/device/floor_painter
+	name = "floor painter"
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "paint sprayer2"
```

Comment:
Not a fan of all these space-including icon states. While I think they should be fine, the convention is generally using underscores. Please.